lib/wallet.js

```diff
@@ -0,0 +1,96 @@
+import React, { useCallback, useContext, useMemo } from 'react'
+import env from './environment'
+import { UnsupportedChainError, UseWalletProvider, useWallet } from 'use-wallet'
+import { providers } from 'ethers'
+import { getNetworkName } from './web3-utils'
+import { trackEvent } from 'lib/analytics'
+
+const CHAIN_ID = Number(env('CHAIN_ID'))
+const WalletAugmentedContext = React.createContext()
+
+function logError(err, ...messages) {
+  if (typeof window !== 'undefined') {
```

Comment:
Just wondering, do we need to do this for SSR?

Reply:
Yup

Reply:
Cool—can we leave a comment? How does this even get called when in SSR?

Reply:
Wasn't my original code but AFAIK it'll try to access `alert` on the `window` object and bark when it's undefined. I'll leave a comment.